client-src/elements/chromedash-link.js

```diff
@@ -319,72 +321,169 @@ function enhanceGoogleDocsLink(featureLink) {
+    this.ignoreHttpErrorCodes = [];
+    /** @type {boolean} */
+    this.alwaysInTag = false;
   }
-  if (!featureLink.information) {
-    if (featureLink.http_error_code &&
-       !ignoreHttpErrorCodes.includes(featureLink.http_error_code)) {
-      return html`<div class="feature-link">
-        <sl-tag>
-          <sl-icon library="material" name="link"></sl-icon>
-          <sl-badge size="small" variant="${featureLink.http_error_code >= 500 ? 'danger' : 'warning'}">
-            ${featureLink.http_error_code}
-          </sl-badge>
-          ${fallback}
-        </sl-tag>
-      </div>`;
+
+  willUpdate(changedProperties) {
+    if (changedProperties.has('href') || changedProperties.has('featureLinks')) {
+      this._featureLink = this.featureLinks.find(fe => fe.url === this.href);
     }
-    return fallback;
   }
-  if (!text) {
-    text = featureLink.url;
+
+  fallback() {
+    const slot = html`<slot>${this.href}</slot>`;
+    return html`<a
+      href=${ifDefined(this.href)}
+      target="_blank"
+      rel="noopener noreferrer"
+      class=${this.class}
+    >${this.alwaysInTag ? html`<sl-tag>${slot}</sl-tag>` : slot}</a>`;
   }
-  try {
-    switch (featureLink.type) {
-      case LINK_TYPE_CHROMIUM_BUG:
-        return enhanceChromeStatusLink(featureLink);
-      case LINK_TYPE_GITHUB_ISSUE:
-        return enhanceGithubIssueLink(featureLink);
-      case LINK_TYPE_GITHUB_PULL_REQUEST:
-        // we use github issue api to get pull request information,
-        // the result will be the similar to github issue
-        return enhanceGithubIssueLink(featureLink);
-      case LINK_TYPE_GITHUB_MARKDOWN:
-        return enhanceGithubMarkdownLink(featureLink);
-      case LINK_TYPE_MDN_DOCS:
-        return enhanceMDNDocsLink(featureLink);
-      case LINK_TYPE_GOOGLE_DOCS:
-        return enhanceGoogleDocsLink(featureLink);
-      case LINK_TYPE_MOZILLA_BUG:
-        return enhanceMozillaBugLink(featureLink);
-      case LINK_TYPE_WEBKIT_BUG:
-        return enhanceWebKitBugLink(featureLink);
-      case LINK_TYPE_SPECS:
-        return enhanceSpecsLink(featureLink);
-      default:
-        return fallback;
+
+  render() {
+    if (!this.href) {
+      console.error('Missing [href] attribute in', this);
+      return html`<slot></slot>`;
+    }
+
+    const featureLink = this._featureLink;
+    if (!featureLink) {
+      return this.fallback();
+    }
+    if (!featureLink.information) {
+      if (featureLink.http_error_code &&
+        !this.ignoreHttpErrorCodes.includes(featureLink.http_error_code)) {
+        return html`<sl-tag>
+            <sl-icon library="material" name="link"></sl-icon>
+            <sl-badge size="small" variant="${featureLink.http_error_code >= 500 ? 'danger' : 'warning'}">
+              ${featureLink.http_error_code}
+            </sl-badge>
+            ${this.fallback()}
+          </sl-tag>`;
+      }
+      return this.fallback();
+    }
+    try {
+      switch (featureLink.type) {
```

Comment:
Switch github to [ignore whitespace](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#diff-view-options) to make most of this diff section go away.